Storage/transaction_policy.h

```diff
@@ -146,19 +149,35 @@ struct Synchronous final {
       return future;                                                                         // LCOV_EXCL_LINE
     }
     bool successful = false;
+    T_RESULT f1_result;
     try {
-      f2(f1());
+      f1_result = f1();
```

Comment:
What if `T_RESULT` does not have a default constructor? Declaring it within the `try` block looks safer to me.